### PR TITLE
Remove legacy symlinks on macOS

### LINKED ISSUE #1984: Remove old integration symlinks on macOS
Currently, `removeLegacySymlinks` is called only for Linux. It is not called for macOS because the old integration symlink directory is `/usr/local/bin/`, which requires root to change. We should figure something out here.

### PR DESCRIPTION
**Pull request by adamkpickering**
Closes #1984.

The idea here is to try to remove legacy symlinks if we can. But if we can't due to a permissions error (which is the expected case on macOS, where the old integration directory is `/usr/local/bin/`, which is owned by root), we want to make a popup that explains that legacy symlinks are there, and that the user should remove them manually.

I haven't tested this on macOS, since I don't have a machine that runs it. However I did test it on my Linux machine and it worked fine.

Finally, I had to do a bit of UI work for this, and I'm not sure I did it right. I think we should provide an answer to the question, "what is a legacy tool symlink?" However, I'm not sure what the best place to put this would be. A tooltip? Just have two paragraphs? @rak-phillip maybe you have a better idea.

### CODE REVIEW

File: src/window/index.ts [hunk 1]
```diff
@@ -252,6 +252,21 @@ export async function openPathUpdate() {
   }));
 }
 
+export async function openLegacyIntegrations(): Promise<void> {
+  const window = openDialog(
+    'LegacyIntegrationNotification',
+    {
+      title:  'This is the title',
```
Comment (rak-phillip) [inline src/window/index.ts:259]: I'd like to suggest a different title for this window, maybe something along the lines of

```suggestion
      title:  'Rancher Desktop - Legacy Integrations',
```

File: src/pages/LegacyIntegrationNotification.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+});
+</script>
+
+<template>
+  <div>
+    <h3>{{ t('legacyIntegrations.title') }}</h3>
+    <p>
+      {{ t('legacyIntegrations.messageFirstPart') }}
+      <code>{{ oldIntegrationPath }}</code>
+      {{ t('legacyIntegrations.messageSecondPart') }}
```
Comment (rak-phillip) [inline src/pages/LegacyIntegrationNotification.vue:30]: A line break to separate the first and second part of the message might help to better communicate that an action is required on the user's part.

File: src/assets/translations/en-us.yaml [hunk 3]
```diff
@@ -277,6 +277,14 @@ pathManagement:
       description: 'Rancher Desktop does not change your PATH configuration; add <code>~/.rd/bin</code> to your path manually.'
   accept: Accept
 
+legacyIntegrations:
+  title: Rancher Desktop - Legacy Integrations Found
```
Comment (rak-phillip) [inline src/assets/translations/en-us.yaml:281]: I think we could drop the `Rancher Desktop -` if we are using this as the headline of the modal. If we intend this to be the window title, we might need two entries to address each case.

File: src/pages/LegacyIntegrationNotification.vue [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    close() {
+      window.close();
+    }
+  },
+});
+</script>
+
+<template>
+  <div>
+    <h3>{{ t('legacyIntegrations.title') }}</h3>
```
Comment (rak-phillip) [inline src/pages/LegacyIntegrationNotification.vue:26]: I think you're on the right track with providing more information and a `v-tooltip` in the header might make sense to help answer some questions. With the current messaging, I might be a bit confused as to what I will actually need to remove.

File: background.ts [hunk 5]
```diff
@@ -160,6 +158,16 @@ Electron.app.whenReady().then(async() => {
       await window.openPathUpdate();
     }
 
+    try {
+      await removeLegacySymlinks(paths.oldIntegration);
+    } catch (error) {
+      if (error instanceof PermissionError) {
+        await window.openLegacyIntegrations();
```
Comment (rak-phillip) [inline background.ts:165]: I'm wondering if we want to alert each time rancher desktop is opened if the user doesn't take any action to clean things up. What do you think?

**Review by rak-phillip — COMMENTED** [review]
Looking great, just left a few comments to get started

**Comment by adamkpickering** [comment]
Note to self: this will probably require adjustment once #2039 is merged.

File: src/window/index.ts (hunk 1 above)
Comment (adamkpickering) [reply src/window/index.ts:259]: Whoops... I remember writing that and thinking "don't forget to fix this!"

File: background.ts (hunk 5 above)
Comment (adamkpickering) [reply background.ts:165]: We could add a checkbox that lets people say "don't show me this again". Which would be more work, and an addition to the settings object.

On the other hand, this popup occurs when somehow the user has manually created symlinks in `/usr/local/bin`, and `/usr/local/bin` is not owned by their user. This isn't going to be a common thing.

In that light, I'd argue that this isn't worth the time/complexity cost. But I'm open to ideas as always :)

Comment (rak-phillip) [reply background.ts:165]: I think we're fine without the checkbox as long as it's clear what action the user needs to take in order to get rid of the prompt.

File: src/pages/LegacyIntegrationNotification.vue (hunk 4 above)
Comment (adamkpickering) [reply src/pages/LegacyIntegrationNotification.vue]: Ok I added a tooltip. Let me know what you think!

**Review by rak-phillip — APPROVED** [review]
LGTM - I would like to see auto resize working, but I think this does enough to convey the message, which is most important.

**Pull request merged**